src/main/scala/nodes/util/CommonSparseFeatures.scala

```diff
@@ -10,17 +10,31 @@ import scala.reflect.ClassTag
  * An Estimator that chooses the most frequently observed sparse features when training,
  * and produces a transformer which builds a sparse vector out of them
  *
+ * Deterministically orders the feature mappings first by decreasing number of appearances,
+ * then by earliest appearance in the RDD
+ *
  * @param numFeatures The number of features to keep
  */
 case class CommonSparseFeatures[T : ClassTag](numFeatures: Int) extends Estimator[Seq[(T, Double)], SparseVector[Double]] {
   // Ordering that compares (feature, frequency) pairs according to their frequencies
-  val ordering = new Ordering[(T, Int)] {
```

Comment:
Ugh, right. You'd think "lexicographic" would be a sensible way to order Seq's for `Seq[T <:< Ordering]` - we could do something like this: http://stackoverflow.com/questions/3137918/how-do-i-sort-a-collection-of-lists-in-lexicographic-order-in-scala - but I digress..

The main thing I'm worried about is the stability of this solution - e.g. whether .zipWithUniqueId is guaranteed to give the same results on the same RDD forevermore, but the docs are convincing enough, I guess.